aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -246,27 +314,77 @@ inline bool check_tensor_shapes(sdp_params params, bool debug) {
   return true;
 }
 
-inline bool check_equal_batch_size_and_num_heads(sdp_params params, bool debug) {
+inline bool check_batch_size_and_num_heads(sdp_params params, bool debug) {
   // This is expected to be called after check_tensor_shapes ensuring that the size()
   // calls won't error since the inputs are all 4 dimensional
-  bool same_batch_size = params.query.size(0) == params.key.size(0) &&
-      params.query.size(0) == params.value.size(0);
-  // We pass through for NestedTensors since this is checked in a later filter
-  bool same_num_heads = params.query.is_nested()
-      ? true
-      : params.query.size(1) == params.key.size(1) &&
-          params.query.size(1) == params.value.size(1);
+  auto q_batch_size = params.query.size(0);
+  auto k_batch_size = params.key.size(0);
+  auto v_batch_size = params.value.size(0);
+
+  // Flip because check_for_nested_inputs returns false if there exists a nested input
+  bool has_nested_input = !check_for_nested_inputs(params);
```

Comment:
Nit: since we are just using this as a helper now and not a filter_function we could probs switch this logic, but no worries